components/calendar/src/types.rs

```diff
@@ -451,20 +497,23 @@ impl ValidMonthCode {
         }
     }
 
-    /// Create a new ValidMonthCode without checking that the number is between 1 and 99
-    #[inline]
+    // precondition: number <= 99
     pub(crate) const fn new_unchecked(number: u8, is_leap: bool) -> Self {
         debug_assert!(1 <= number && number <= 99);
         Self { number, is_leap }
     }
 
-    /// Returns the month number according to the month code.
+    /// Returns the month number.
+    ///
+    /// A month number N is not necessarily the Nth month in the year if there are leap
+    /// months in the year, rather it is associated with the Nth month of a "regular"
```

Comment:
issue: this definition is incorrect: Adar I is M05L but it is associated with Adar II, which is M06.

The definition of month number the way it is used in month codes is, unfortunately, "for non leap months the month number of that month in a regular year, and for leap months the month number of the month _after_ it in a regular year."

I was opposed to this definition, but it's what the standard decided on. If we want to have the "associated with the Nth month of a regular year" definition we cannot have this produce a month code.

Unless we store `leap` as a tri-value `NotLeap, BeforeRegular, AfterRegular` enum. Then that would work, and it would probably be a better API, but it makes `new_leap` confusing.

Reply:
I prefer to delete the part of the sentence starting with "rather", and tell people to go look at RFC 7529 or ECMA 262 if they have questions.

I really really don't want to innovate on month representation. I'm okay with a "type-safe month code" but I don't have an appetite for more than that.

Reply:
Neither RFC 7528 nor ECMA 262 have any mentions of month code, can you clarify what you want me to link to?

Reply:
We need to explain this somehow, otherwise how are callers going to know that Adar I is `Month::leap(5)` and not `Month::leap(6)`?

> The definition of month number the way it is used in month codes is, unfortunately, "for non leap months the month number of that month in a regular year, and for leap months the month number of the month after it in a regular year."

This is for Hebrew, right? In EastAsianTraditional, `M02L` is after `M02`, no?

Reply:
Don't try to make any relationship between a particular leap month and a particular common month. The invariant is the ordering of codes, so let's lean into that definition. You can say:

> In calendars with leap months, a month with number N is not necessarily the Nth month in the year, and there may be multiple months with number N in the same year. A leap month with number N occurs after the non-leap month with number N, if it exists, and before the non-leap month with number N+1, if it exists.